python/lsst/pipe/base/graph/_loadHelpers.py

```diff
@@ -219,7 +220,11 @@ def load(
         _readBytes = self._readBytes
         if universe is None:
             universe = headerInfo.universe
-        return self.deserializer.constructGraph(nodeSet, _readBytes, universe)
+        # use the daf butler context vars to aid in ensuring deduplication in
+        # object instantiation.
+        runner = PersistenceContextVars()
+        graph = runner.run(self.deserializer.constructGraph, nodeSet, _readBytes, universe)
+        return graph  # type: ignore
```

Comment:
I do not entirely understand PersistenceContextVars/context vars, would it be nice if it could work like a regular context manager, e.g.:

    with PersistenceContextVars():
       return self.deserializer.constructGraph(nodeSet, _readBytes, universe)

but may be it's not possible?

Reply:
I worked on that for a while, and the outcome kept coming out more awkward, and not at all nice like that. I think it is mostly because context vars are more geared toward asyncio, callback programming style.